book-tr/svg_lib.py

```
@@ -0,0 +1,666 @@
+# Shrinks any <text> that overflows its smallest containing <rect> or the
+# canvas. Only the font-size is changed (never positions), so it is safe and
+# idempotent. Used both by SVG.render() (so generated figures self-correct
+# standalone labels) and by fit_svg_text.py (to repair static/orphaned SVGs
+# that no longer have a generator).
+_TEXT_TAG = re.compile(r'<text\b([^>]*)>(.*?)</text>', re.S)
+_RECT_TAG = re.compile(
+    r'<rect\b[^>]*?x="([-\d.]+)"[^>]*?y="([-\d.]+)"[^>]*?'
+    r'width="([-\d.]+)"[^>]*?height="([-\d.]+)"'
+)
+_VIEWBOX = re.compile(r'viewBox="([-\d.]+) ([-\d.]+) ([-\d.]+) ([-\d.]+)"')
+_ATTR = lambda attrs, name: (re.search(name + r'="([^"]*)"', attrs) or [None, None])[1]
```

Ruff (0.15.21)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `_ATTR` as a `def`

(E731)